lib/src/main/java/uk/gov/di/ipv/core/library/helpers/RequestHelper.java

```diff
@@ -98,21 +104,50 @@ public static String getClientOAuthSessionId(APIGatewayProxyRequestEvent event)
         return getClientOAuthSessionId(event.getHeaders());
     }
 
-    private static String getIpvSessionId(Map<String, String> headers, boolean allowNull)
+    public static String getIpvSessionId(JourneyRequest request, boolean allowNull)
+            throws HttpResponseExceptionWithErrorBody {
+        String ipvSessionId = request.getIpvSessionId();
+
+        validateIpvSessionId(ipvSessionId, "ipvSessionId not present in request", allowNull);
+
+        LogHelper.attachIpvSessionIdToLogs(ipvSessionId);
+        return ipvSessionId;
+    }
+
+    public static String getIpvSessionId(Map<String, String> headers, boolean allowNull)
             throws HttpResponseExceptionWithErrorBody {
         String ipvSessionId = RequestHelper.getHeaderByKey(headers, IPV_SESSION_ID_HEADER);
+        String message = String.format("%s not present in header", IPV_SESSION_ID_HEADER);
+
+        validateIpvSessionId(ipvSessionId, message, allowNull);
+
+        LogHelper.attachIpvSessionIdToLogs(ipvSessionId);
+        return ipvSessionId;
+    }
+
+    private static void validateIpvSessionId(
+            String ipvSessionId, String errorMessage, boolean allowNull)
+            throws HttpResponseExceptionWithErrorBody {
         if (ipvSessionId == null) {
-            String message = String.format("%s not present in header", IPV_SESSION_ID_HEADER);
             if (allowNull) {
-                LOGGER.warn(message);
+                LOGGER.warn(errorMessage);
             } else {
-                LOGGER.error(message);
+                LOGGER.error(errorMessage);
                 throw new HttpResponseExceptionWithErrorBody(
                         HttpStatus.SC_BAD_REQUEST, ErrorResponse.MISSING_IPV_SESSION_ID);
             }
         }
-        LogHelper.attachIpvSessionIdToLogs(ipvSessionId);
-        return ipvSessionId;
+    }
+
+    public static String getIpAddress(JourneyRequest request)
+            throws HttpResponseExceptionWithErrorBody {
+        String ipAddress = request.getIpAddress();
+        if (ipAddress == null) {
+            LOGGER.error("ipAddress not present in request");
+            throw new HttpResponseExceptionWithErrorBody(
```

Comment:
Can have similar validateIpAddress(...) to move duplicate code.

Reply:
Just added